internal/librarian/state.go

```diff
@@ -124,3 +124,12 @@ func findServiceConfigIn(contentLoader func(file string) ([]byte, error), path s
 
 	return "", errors.New("could not find service config in " + path)
 }
+
+func saveLibrarianState(repoDir string, state *config.LibrarianState) error {
```

Comment:
fyi - this seems duplicated/recreated in #989 writeLibrarianState

Reply:
I didn't realize the commit sha needs to change so we should write librarian state here.

Could you remove the similar function in https://github.com/googleapis/librarian/blob/main/internal/librarian/generate.go#L522